_posts/2020-12-19-generate-mini-site-with-metanorma-cli.adoc

```diff
@@ -0,0 +1,161 @@
+then you can run the following:
+
+[source, sh]
+----
+metanorma site generate ./sources --output-dir site-folder
+----
+
+That's it, metanorma will take care of everything now, like compiling those
+source files, building the collection, and finally generating the HTML site.
+To keep you updated, metanorma will also print out the details in the terminal.
+
+These are the very bare minimum to generate a site, but let's look at the help
+to see what's available. Running `metanorma site help generate`:
+
+[source, sh]
+----
+Usage:
+  metanorma site generate SOURCE_PATH
+
+Options:
+  -c, [--config=CONFIG]          # The metanorma configuration file
+  -o, [--output-dir=OUTPUT_DIR]  # Output directory for the generated site
+                                 # Default: /home/[your-repo-path]/site
+----
+
+This should cover most of the use cases, but what if we only want to include
+some specific files or add custom site metadata, well, then the next section is
+for you.
+
+=== Customizing site generation
+
+If you are like us and don't miss out on the details then you might have seen
+the `-config` option in `site generate` help documentation, we didn't say
+anything earlier but that's the option to customize the build.
+
+This option takes a `YAML` file and uses that as site generation instructions,
+Currently it offers the following customizations:
+
+* Build a site with specific source files
+* Customize site metadata (Eg: title, and copyright)
+
+==== Build a site with specifc source files
+
+The demo projects contain a couple of documents in the source directory, but for
+our example's sake let's say we only want to use `sources/G.191.adoc` for the
+site generation, how can we do it?
+
+That's the perfect scenario for a configuration file, the name of the file could
+be anything but please pay attention to the node structure:
+
+[source, ruby]
+----
+# filename: metanorma.yml
+
+metanorma:
+  source:
+    files:
+      - G.191.adoc
+----
+
+At the top level we have `metanorma`, then nested `source`, and then one more
+level of nesting with `files`. In `files` we can provide the relative path (In
+terms of source directory) for each file as a list and then run the following:
+
+[source, sh]
+----
+metanorma site generate ./sources -o site-folder --config metanorma.yml
+----
+
+This will build the site with the specific files you have added in your config
+file, but if you need to be more flexible then we also support wildcard for file
+listing
+
+==== Customize site metadata (Eg: title, copyright)
+
+Now let's look at how can we provide a custom title for the hero area or the
+name for the site copyright. To do that we can use the same configuration file
+and add the following data to it.
+
+[source, sh]
+----
+# filename: metanorma.yml
+
+metanorma:
+  collection:
+    name: Sample Name for the hero area
+    organization: Sample organization name
+----
+
+Then pass this configuration file as earlier to customize your build. You can
```

Comment:
https://github.com/metanorma/metanorma.com/issues/377 says to add metanorma.yml default config path to blog post....

Reply:
I've added these details in the earlier section of the same block.